app/src/main/java/net/kuama/documentscanner/domain/PerspectiveTransform.kt

```diff
@@ -22,43 +21,66 @@ class PerspectiveTransform : UseCase<Bitmap, PerspectiveTransform.Params>() {
     class Params(val bitmap: Bitmap, val corners: Corners)
 
     override suspend fun run(params: Params): Either<Failure, Bitmap> = try {
-        val src = Mat()
+        val sourceBitmapMatrix = Mat()
+        val srcMat = Mat(4, 1, CvType.CV_32FC2)
+        val dstMat = Mat(4, 1, CvType.CV_32FC2)
 
-        Utils.bitmapToMat(params.bitmap, src)
+        Utils.bitmapToMat(params.bitmap, sourceBitmapMatrix)
 
         val orderedCorners = PerspectiveTransformUtils.sortPoints(params.corners)
 
-        val widthA = sqrt(
-            (orderedCorners.bottomRight.x - orderedCorners.bottomLeft.x).pow(2.0) + (orderedCorners.bottomRight.y - orderedCorners.bottomLeft.y).pow(2.0)
-        )
-        val widthB = sqrt(
-            (orderedCorners.topRight.x - orderedCorners.topLeft.x).pow(2.0) + (orderedCorners.topRight.y - orderedCorners.topLeft.y).pow(2.0)
-        )
-        val dw = max(widthA, widthB)
-        val maxWidth = dw.toInt()
-        val heightA = sqrt(
-            (orderedCorners.topRight.x - orderedCorners.bottomRight.x).pow(2.0) + (orderedCorners.topRight.y - orderedCorners.bottomRight.y).pow(2.0)
+        val widthA = AlgebraUtils
+            .calculateDistanceBetweenTwoPoints(orderedCorners.bottomRight, orderedCorners.bottomLeft)
+
+        val widthB = AlgebraUtils
+            .calculateDistanceBetweenTwoPoints(orderedCorners.topRight, orderedCorners.topLeft)
+
+        val maxWidthDouble = max(widthA, widthB)
+
+        val maxWidthInt = maxWidthDouble.toInt()
+
```

Comment:
```suggestion
        val maxWidth = max(widthA, widthB) 
```
I'd keep only one variable (just the double) for max height and width, and if you need the integer just add `.toInt()` where necessary, please apply where needed